Ys2 avd

yshafner opened the pull request:
<!-- Thank you for submitting a Pull Request. Please fill out the template below.-->
# Overview/Summary

Removed Broken Querries! 
I have started updating the AVD page on APRL. We have added 6 recommendations. We have more to add but I want to make sure these look good before continuing. 

## Related Issues/Work Items

[AB#32444](http://localhost:1313/Azure-Proactive-Resiliency-Library/contributing/#azure-resource-graph-arg-queries)

## This PR adds

1. 6 AVD Items
Here are some scene shots of the Querries: 
<img width="932" alt="APRL Private link AVD 1" src="https://github.com/Azure/Azure-Proactive-Resiliency-Library/assets/64444337/d7cdb287-66e3-46f8-ac1e-7b6a9d92bbeb">
<img width="939" alt="APRL Private link AVD 2" src="https://github.com/Azure/Azure-Proactive-Resiliency-Library/assets/64444337/dcaa338b-4a84-429a-84d8-34ec19948aa4">
<img width="932" alt="APRL Public network access AVD 1" src="https://github.com/Azure/Azure-Proactive-Resiliency-Library/assets/64444337/5d5ac896-7827-4a2d-aad7-b65d92bb587b">
<img width="928" alt="APRL Public network access AVD 2" src="https://github.com/Azure/Azure-Proactive-Resiliency-Library/assets/64444337/e97e1a1a-3de6-4a03-9896-c83d9d7fe88b">


## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

oZakari commented on docs/content/services/specialized-workloads/azure-virtual-desktop/_index.md:
```diff
@@ -0,0 +1,191 @@
+
+**Impact: Medium**
+
+**Recommendation/Guidance**
```
This looks to be duplicate recommendation guidance from Recommendation 1. Although these are similar, is there anything specific we can adjust to avoid confusion for people?

ejhenry replied:
Could you also add detail on how 1 & 2 relate to reliability and how following the recommendations improve the workload reliability? The current wording sound more like general security best practices.

oZakari commented on docs/content/services/specialized-workloads/azure-virtual-desktop/_index.md:
```diff
@@ -0,0 +1,191 @@
+
+**Recommendation/Guidance**
+
+Use scaling plans, which automatically turn hosts off and on to help ensure adequate performance for users.
+Adjust the settings of scaling plans to improve cost efficiency.
```
I agree that this is a relevant recommendation, but can you please add a bit more to clarify how this impacts resiliency, specifically?

oZakari commented on docs/content/services/specialized-workloads/azure-virtual-desktop/code/avd-1/avd-1.kql:
```diff
@@ -0,0 +1,4 @@
+resources
```
For both working queries, can you add a description of what they are doing at the top as shown here: http://localhost:1313/Azure-Proactive-Resiliency-Library/contributing/#azure-resource-graph-arg-queries

ejhenry reviewed (commented):
Recommendations need to be written in a clear, concise manner to articulate the benefits to workload reliability. Please update recommendations 1, 2, and 5 to align with this requirement.

yshafner commented:
> Hi @yshafner, thanks for creating a new PR! I added some notes for some potential changes/considerations.
> 
> Also, I added a commit to fix some general lint errors as an heads up. Otherwise, feel free to ping me if you have any questions.
> 
> Also, I see that you added a checkmark for an Azure DevOp work item, but don't see it linked in the PR form. Do you have one that you can add?

@oZakari Here is the Dev ops work item: https://dev.azure.com/CSUSolEng/Accelerator%20-%20AVD/_workitems/edit/32444
@ejhenry I have made the requested changes and submitted the request. 

Please let me know if there are any other needed changes!

oZakari commented:
Discussed with @yshafner about some additional changes/additions to the recommendations. He will connect with his team and we may need to have align with the broader AVD team.

cc: @ejhenry

oZakari reviewed (approved):
LGTM, thanks @yshafner!

oZakari commented:
@ejhenry could you take a look and approve this when you get a chance? This will be the first iteration for AVD.

ejhenry reviewed (approved):
LGTM 👍
